### DISCUSSION TITLE
separate audio mixture only

### DISCUSSION THREAD

**Discussion by taalua** [hmartelb/avlit#3] [General]
Thank you for the awesome work.
Is it possible to separate the input audio mixture only without video frames? (maybe by setting the video path to null).

Thank you

**Comment by hmartelb** [comment]
Hello! Thank you for your question. 
It is theoretically possible to use only the audio mixture, but you should expect the separation performance to be worse. 

You can use one of the following options to achieve this:
1. Modify the [forward method](https://github.com/hmartelb/avlit/blob/010a4720b4d342d68ac7d3d4e9a5a7645f007a0c/src/avlit.py#L121) to set `v: torch.Tensor = None` as default so you only need to pass the audio mixture. Then, check if `v` is defined before calling the video encoder and video branch, skip these modules if it is `None`. 
2. Pass a dummy tensor `v` containing zeros/ones with the right shape. This requires no modification to the code, but will result in more computation than strictly necessary due to the frame encoder. You can also set `video_num_blocks = 0` to avoid using the video branch.

I hope this helps.
